shared_model/interfaces/iroha_internal/transaction_sequence.hpp

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_TRANSACTION_SEQUENCE_HPP
+#define IROHA_TRANSACTION_SEQUENCE_HPP
+
+#include "common/result.hpp"
+#include "interfaces/common_objects/types.hpp"
+
+namespace shared_model {
+  namespace interface {
+
+    namespace types {
+      using TransactionForwardCollectionType =
+          boost::any_range<Transaction,
+                           boost::forward_traversal_tag,
+                           const Transaction &>;
+    }
+
+    /**
+     * Transaction sequence is the collection of transactions where:
+     * 1. All transactions from the same batch are place contiguously
```

Comment:
Those rules are related to validation. Maybe better to move it to validation class?